### PR TITLE
Added FloatEuclidTransformVectorList class.

### PR DESCRIPTION
**Pull request by niklasborwell**
Basically just added this class to simplify the conversion from VectorList\<EuclidTransform\> to FloatVectorLists containing the translations and rotations. This is currently done in several scattered places using private classes.
Also added of x y and z properties in FloatRotation3D that are needed.

### CODE REVIEW

**Comment by niklasborwell** [comment]
@marcusnaslund Maybe you can peer review this? Or push it to someone else. It is very similar to the FloatPoint2DVectorList class, with FloatEuclidTransforms instead of FloatPoint2Ds.

**Comment by marcusnaslund** [comment]
I will take it.

File: source/math/FloatEuclidTransformVectorList.ooc [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+*/
+use ooc-collections
+import math
+import FloatEuclidTransform
+import FloatRotation3D
+import FloatVectorList
+
+FloatEuclidTransformVectorList: class extends VectorList<FloatEuclidTransform> {
+	init: func ~default {
+		this super()
```
Comment (marcusnaslund) [inline source/math/FloatEuclidTransformVectorList.ooc]: No `this` before `super`.

File: source/math/FloatEuclidTransformVectorList.ooc [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+	}
+	init: func ~fromVectorList (other: VectorList<FloatEuclidTransform>) {
+		this super(other _vector)
```
Comment (marcusnaslund) [inline source/math/FloatEuclidTransformVectorList.ooc]: No `this` before `super`.

File: source/math/FloatEuclidTransformVectorList.ooc [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+	getScaling: func -> FloatVectorList {
+		result := FloatVectorList new()
+		for (i in 0 .. this _count) {
+			euclidTransform := this[i]
+			result add(euclidTransform scaling)
+		}
+		result
+	}
+	operator [] (index: Int) -> FloatEuclidTransform {
+		this _vector[index] as FloatEuclidTransform
```
Comment (marcusnaslund) [inline source/math/FloatEuclidTransformVectorList.ooc]: Do you really need to cast?

File: source/math/FloatRotation3D.ooc [hunk 4]
```diff
@@ -22,6 +22,9 @@ FloatRotation3D: cover {
 	inverse ::= This new(this _quaternion conjugate) //FIXME Should this be conjugate or inverse?
```
Comment (marcusnaslund) [inline source/math/FloatRotation3D.ooc]: Not related to this PR, but could you look this up and fix when you have time?

File: source/math/FloatEuclidTransformVectorList.ooc [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+	toVectorList: func -> VectorList<FloatEuclidTransform> {
+		result := VectorList<FloatEuclidTransform> new()
+		result _vector = this _vector
+		result _count = this _count
+		result
+	}
+	getTranslationX: func -> FloatVectorList {
+		result := FloatVectorList new()
+		for (i in 0 .. this _count) {
+			euclidTransform := this[i]
```
Comment (marcusnaslund) [inline source/math/FloatEuclidTransformVectorList.ooc]: You decide if you want a `thisPointer` thing for these functions.

**Comment by marcusnaslund** [comment]
Done reviewing. Thank you for adding the test as well.

File: source/math/FloatEuclidTransformVectorList.ooc (hunk 1 above)
Comment (niklasborwell) [inline source/math/FloatEuclidTransformVectorList.ooc]: This is what happens when you copy another class....same thing in FloatPoint2DVectorList. I'll fix it.

Comment (marcusnaslund) [inline source/math/FloatEuclidTransformVectorList.ooc]: Yep, I just fixed it there. :)

File: source/math/FloatEuclidTransformVectorList.ooc (hunk 5 above)
Comment (niklasborwell) [inline source/math/FloatEuclidTransformVectorList.ooc]: I'll leave it until we need it.

File: source/math/FloatRotation3D.ooc (hunk 4 above)
Comment (niklasborwell) [inline source/math/FloatRotation3D.ooc]: Yes, we should have a look at this.

**Comment by niklasborwell** [comment]
Updated.

File: source/math/FloatRotation3D.ooc [hunk 6]
```diff
@@ -22,6 +22,9 @@ FloatRotation3D: cover {
...
 	normalized ::= This new(this _quaternion normalized)
 	transform ::= this _quaternion transform
+	x ::= this _quaternion rotationX
+	y ::= this _quaternion rotationY
+	z ::= this _quaternion rotationZ
```
Comment (marcusnaslund) [inline source/math/FloatRotation3D.ooc]: You cannot run the tests in the other project with this, unfortunately.

Comment (niklasborwell) [inline source/math/FloatRotation3D.ooc]: Fixed here: https://github.com/vidhance/ooc-vidproc/pull/1063

Comment (marcusnaslund) [inline source/math/FloatRotation3D.ooc]: I think we had a discussion about linking to private repos before? :)

**Comment by niklasborwell** [comment]
Updated.

File: source/math/FloatRotation3D.ooc (hunk 6 above)
Comment (niklasborwell) [inline source/math/FloatRotation3D.ooc]: Just saw it before I saw your message >_<

**Pull request merged**